WorkflowUI/Sources/ViewControllerDescription/ViewControllerDescription.swift

```diff
@@ -46,31 +59,46 @@
             }
         }
 
-        /// Construct and update a new view controller as described by this view
-        /// controller description.
-        internal func buildViewController() -> UIViewController {
+        /// Construct and update a new view controller as described by this view controller description.
+        /// The view controller will be updated before it is returned, so it is fully configured and prepared for display.
+        public func buildViewController() -> UIViewController {
             let viewController = build()
-            assert(canUpdate(viewController: viewController), "View controller description built a view controller it cannot update (\(viewController) is not exactly type \(viewControllerType))")
 
             // Perform an initial update of the built view controller
             update(viewController: viewController)
 
             return viewController
         }
 
-        /// If the given view controller is of the correct type to be updated by
-        /// this view controller description.
-        internal func canUpdate(viewController: UIViewController) -> Bool {
+        /// If the given view controller is of the correct type to be updated by this view controller description.
+        ///
+        /// If your view controller type can change between updates, call this method before invoking `update(viewController:)`.
+        ///
+        /// ### Note
+        /// Failure to confirm the view controller is updatable will result in a fatal `precondition`.
+        public func canUpdate(viewController: UIViewController) -> Bool {
             return type(of: viewController) == viewControllerType
         }
 
-        /// Update the given view controller.
+        /// Update the given view controller with the content from the view controller description.
         ///
-        /// - Note: Passing a view controller that does not return `true` from
-        ///         `canUpdate(viewController:)` will result in an exception.
+        /// - Parameters:
+        ///   - viewController: The view controller to update.
         ///
-        /// - Parameter viewController: The view controller instance to update
-        internal func update(viewController: UIViewController) {
+        /// ### Note
+        /// You must pass a view controller that is exactly the same type as the type passed to `init`'s `type`
```

Comment:
I'd prefer to keep this a little more implementation-vague with something like 

> You must pass a view controller previously created by a compatible ViewControllerDescription and that passes `canUpdate(viewController:)`. Failure to do so will result in a fatal `precondition`.

rather than talking about the type itself (slightly leaving the door open for further refinements internally here if they're ever needed).

Reply:
Done!